File: support_tools/swigwin-3.0.12/Examples/test-suite/perl5/array_member_runme.pl
#!/usr/bin/perl
use strict;
use warnings;
use Test::More tests => 18;
BEGIN { use_ok('array_member') }
require_ok('array_member');

my $f = array_member::Foo->new();
$f->{data} = $array_member::global_data;

for(my $i=0; $i<8; $i++) {
	is( array_member::get_value($f->{data},$i),
		array_member::get_value($array_member::global_data,$i),
		"array assignment");
}

for(my $i=0; $i<8; $i++) {
	array_member::set_value($f->{data},$i,-$i);
}

$array_member::global_data = $f->{data};

for(my $i=0; $i<8; $i++) {
	is(array_member::get_value($f->{data},$i),
		array_member::get_value($array_member::global_data,$i),
		"array assignment");
}

File: support_tools/swigwin-3.0.12/Examples/test-suite/octave/array_member_runme.m
array_member

f = Foo();
f.data = cvar.global_data;

for i=0:7,
    if (get_value(f.data,i) != get_value(cvar.global_data,i))
      error("Bad array assignment");
    endif
endfor

for i=0:7,
    set_value(f.data,i,-i);
endfor

cvar.global_data = f.data;

for i=0:7,
  if (get_value(f.data,i) != get_value(cvar.global_data,i))
    error("Bad array assignment")
  endif
endfor



